File: Easy/495. Teemo Attacking/Solution.java
class Solution {
    public int findPoisonedDuration(int[] timeSeries, int duration) {
        int total = 0;
        for (int i = 1; i < timeSeries.length; i++) {
            if (timeSeries[i] - timeSeries[i - 1] <= duration) {
                total += timeSeries[i] - timeSeries[i - 1];
            } else {
                total += duration;
            }
        }
        return total;

    }
}

File: Easy/495. Teemo Attacking/Solution.py
class Solution(object):
    def findPoisonedDuration(self, timeSeries, duration):
        """
        :type timeSeries: List[int]
        :type duration: int
        :rtype: int
        """
        if len(timeSeries) == 1:
            return duration
        
        total = 0
    
        for i in range (len(timeSeries) -1):
             if (timeSeries[i] + duration >= timeSeries[i + 1]) :
                total += timeSeries[i + 1] - timeSeries[i];
             else :
                total += duration;
            
        total += duration;
        return total;